### ISSUE TITLE
write_csv()

### ISSUE THREAD

**Issue by alexhallam** [pola-rs/polars#2699]
|word | compliment |
|---------|--------------|
| read | write |
| from | to |

Yet this package uses the language `read_csv` and `to_csv`. 

It would be nice if `to_csv` was changed to `write_csv`. 

The language is more natural.

**Comment by ghuls** [comment]
pandas also used `to_`, while pyarrow uses `write_`:

As the other converters are also with`to_` it keeps sense to keep it as it is.
```
In [34]: pl.DataFrame.to_
                          to_arrow()   to_avro()    to_csv()     to_dict()    to_dicts()   to_dummies() to_ipc()     to_json()    to_numpy()   to_pandas()  to_parquet() to_series()

In [33]: df_pd.to_
 to_clipboard() to_dict()      to_feather()   to_hdf()       to_json()      to_markdown()  to_parquet()   to_pickle()    to_sql()       to_string()    to_xarray()
 to_csv()       to_excel()     to_gbq()       to_html()      to_latex()     to_numpy()     to_period()    to_records()   to_stata()     to_timestamp() to_xml()

In [33]: pa.csv.
                 ConvertOptions     InvalidRow         ParseOptions       write_csv()
                 CSVStreamingReader ISO8601            read_csv()         WriteOptions
                 CSVWriter          open_csv()         ReadOptions

In [33]: pa.feather.
                     check_chunked_overflow() ext                      os                       schema()                 write_feather()
                     Codec                    FeatherDataset           read_feather()           Table
                     concat_tables()          FeatherError             read_table()             Version
```

although I might be an option to add `write_csv`, `write_ipc` `write_avro` as aliases.

**Comment by slonik-az** [comment]
Might be changing `read_csv => from_csv ` would be a better choice to restore English language consistency   😄

**Comment by universalmind303** [comment]
I think `to` is more accurate in some scenarios as you are not always performing a `write` action. 

```py
# i suppose you could argue that you are writing to a string though.
csv_str = df.to_csv()
print(csv_str)
```
I agree that the usage of `read/to` is inconsistent though, and either `read/write` or `from/to` would be more natural.

**Comment by ritchie46** [comment]
Yeah.. I am also more inclined to use `from_*`

**Comment by alexhallam** [comment]
Methods should be verbs (`read`,`write`,`cast`).

Prepositions (`from`,`to`,`at`, `on`,`below`,`above`) are not as well suited to the task. There are not as many prepositions as there are verbs.  Also, verbs are more descriptive.

I would argue that for any method that is named with a preposition there is a verb that can describe the method just as well if not better than the preposition.

**Comment by alexhallam** [comment]
I know that is how things _are_. How a thing _is_ and _should be_  are not always the same. 

I am not here to also point at Rust naming conventions as a whole. I would say that `into_iter` and `to_string` could also have better alternatives. Also, why is one thing `into` while another is `to`, `into_string` or `to_iter` are _nearly_ fungible with `to_string` and `into_iter`. To me that is saying the their original naming was not that unique or descriptive in the first place. I agree with you. The toothpaste has already left the tube. No sense in going back on Rust naming as a whole.I am only saying that there is an actionable opportunity here in this package `to_csv()` should be changed to `write_csv()`. 
1. It is the compliment of `read_*`
2. It describes the most common use of the method. Writing the file.

Also, I think `read_*` is better than `from`. This is because `from` can be many things. In Rust when `from` is used it is used to `cast` or `convert`. Here, the operation is no a cast it is a literal reading of a file. Why trade a specific word for a less specific one?

You do not have to agree with me. I opened the issue to show that from my perspective the naming is awkward. 

It is always fine to disagree and move forward. I thank you all for listening. I feel heard and my perspective is out there. You can decided to do with it what you want.

**Comment by ritchie46** [comment]
Ok, ok I am convinced ;). Let's rename the `to_file` methods to `write_file` and deprecate the old names.

**Comment by ritchie46** [comment]
Closed by #2916
